code/modules/awaymissions/maploader/reader.dm

```diff
@@ -304,6 +311,16 @@ var/list/map_dimension_cache = list()
 	members.Remove(members[index])
 
 	//then instance the /turf and, if multiple tiles are presents, simulates the DMM underlays piling effect (only the last turf is spawned, other ones are drawn as underlays)
+	if(overwrite) //make this come first so lighting overlays don't die
+		var/turf/T_old = locate(xcrd,ycrd,zcrd)
+		var/static/list/blacklisted_types = list(/mob/dead/observer,/mob/dview,/atom/movable/lighting_overlay)
+		var/list/truncated_contents = T_old.contents.Copy()
+		while(truncated_contents.len)
+			truncated_contents = T_old.contents.Copy()
+			for(var/type in blacklisted_types)
+				truncated_contents = prune_list_to_type(truncated_contents,type,TRUE)
```

Comment:
Instead of iterating over the contents `blacklisted_types.len` times, you could do this in one go and use `is_type_in_list`.

However, looking at this from a higher level perspective, you're still trying to solve the problem of deleting all objects in one tile, which SHOULD be as simple as a single for-loop (without the `while`). The reason that's not sufficient, as far as I can guess, is that some objects create other objects on the tile they were in whenever they're destroyed. Instead of letting such objects do that and cleaning up after them, I think the correct approach would be to prevent them from creating the objects in the first place.

Reply:
To my knowledge this is because of closets creating their contents and vending machines creating the coinbox, those fixes don't feel like an exhaustive list as there could be even more things to check.

Reply:
I feel like they should be fairly simple problems to solve. If something is being *destroyed* (due to damage or something), make them drop the contents. If they're being *deleted* but they haven't been destroyed, don't.